### DISCUSSION TITLE
Balls keep rolling forever

### DISCUSSION THREAD

**Discussion by michaelbortolin** [bepu/bepuphysics2#339] [Q&A] [answered]
I can't seem to make balls stop rolling after they start moving. I have tried different friction coefficients for both ball and ground, but it does not make any difference. This behaviour is also seen in the "BouncinessDemo"... 

What am I missing?

**Comment by RossNordby** [comment, marked as answer]
Friction opposes relative sliding motion on a contact surface. A sphere rolling on a surface can have zero relative velocity at the contact point, so friction cannot stop rolling by itself. (Or, equivalently, friction is why a ball slid across a table will start to roll instead.)

There is also a concept of "rolling friction," but that works differently. It arises from material deformation and imperfections that aren't uniformly present in the simulation by default. Twist friction actually does model this; that's why a spinning sphere squeezed tightly by two planes will stop spinning faster than one gently resting on a surface. I *could* handle this as a part of friction for rolling, too, but it adds a bit of cost and isn't always desirable, so I didn't bake it in.

An easy way to introduce an approximation of rolling friction is to instead use damping. The `BouncinessDemo` explicitly disables damping because it wants to demonstrate bounciness; see the constructor of the `DemoPoseIntegratorCallbacks` passed into `Simulation.Create`. Increasing either linear or angular damping would make the balls stop rolling. If you want to stop rolling without affecting linear motion, angular damping would be the better target. Notably, damping isn't a fancy built-in feature, it's just a simple exponential decay of momentum handled by the `DemoPoseIntegratorCallbacks` that you could handle yourself however you'd like.
